client/modules/module.js

```diff
@@ -129,85 +142,78 @@ export class ClientModule {
       peerNetwork,
       assert,
     };
-    const {load} = await import(this.path);
-    if (!load) {
-      throw new Error(`${this.name} did not export a 'load' function!`);
-    }
-    const {client} = inject(load, fakeEnv);
-    conform(client, moduleInterface.Client);
-
-    this.instance = new client(this.config);
-  }
-
-  willBeHiddenSoon() {
-    if (!this.path) {
-      return true;
-    }
     try {
-      this.instance.willBeHiddenSoon();
-    } catch(e) {
-      error(e);
+      const {load} = await ClientModule.loadPath(this.path);
+      if (!load) {
+        throw new Error(`${this.name} did not export a 'load' function!`);
+      }
+      const {client} = inject(load, fakeEnv);
+      conform(client, moduleInterface.Client);
+
+      this.instance = new client(this.config);
+    } catch (e) {
+      // something went very wrong. Wind everything down.!
+      this.network.close();
+      this.network = null;
+      throw e;
     }
-    return true;
   }
 
   // Returns true if module is still OK.
-  willBeShownSoon() {
+  async willBeShownSoon() {
     if (!this.path) {
```

Comment:
Smells like strategy or null-object pattern is wanted here, like rather than guarding every method on this.path, there'd be wrapper thing that either noops each method or "does the stuff". Could be overkill though.

Reply:
Yeah. I think I'd like to unify the module_interfaces first (soon!) before doing such a thing. But yes.

Reply:
SG. I'll look through the whole thing again with a more careful eye, as well.